### PR TITLE
Added devise to my project

### PR DESCRIPTION
**Pull request by ismayito**
### Add devise to project and customize sign up and login views
_______________________________________________________________________________________________________
#### Summary:
_______________________________________________________________________________________________________In this project we added the following
- Installed and setup devise
- Made a user to register as a new user
- Made a user to be able to login after sign up
- A user can aks for confirmation of email
- A user can reset a password

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,13 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (TedLivist) [inline app/models/user.rb]: - [ ] You are repeating code here. Kindly remove one.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,13 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (TedLivist) [inline app/models/user.rb]: - [ ] Users should be required to confirm their emails after signing up before logging in. Kindly add the missing `:confirmable` module here.

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<h2>Welome! please Log in if you have an account already else Sign up ?</h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name),class:'login-form') do |f| %>
+ <div class="form-group">
+  <div class="field-input">
+    <%= f.email_field :email, autofocus: true, placeholder:'Username', autocomplete: "email" %>
```
Comment (TedLivist) [inline app/views/devise/sessions/new.html.erb]: - [ ] The placeholder here should be `email`, not `Username` for a better user experience

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ismayito and @Uzair-Manzoor,

Good job so far!
✔️ Integration of Devise into Rails app
✔️ Redesigning of forms to fit wireframe
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+  def configure_sign_in_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_in, keys: [:Name])
+  end
+  # helper_method :current_user
+
+  # private
+
+  # def current_user
+  #   @current_user ||= User.first
+  # end
```
Comment (TedLivist) [inline app/controllers/application_controller.rb]: - [ ] Kindly remove commented code here also.

File: app/models/user.rb [hunk 5]
```diff
@@ -1,4 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (LionRouge1) [inline app/models/user.rb:5]: - [ ] Nice work making the changes requested by the previous reviewer :clap: However, your App raise an error (see image) when registering a new user. Kindly make sure a user can register successfully without any error.
Screenshot from 2023-12-21 14-35-08

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @ismayito, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

## Suggested changes :recycle: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by aamirkhan2478 — APPROVED** [review]
Hi @ismayito , 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly :+1: 
✅ No linter error :+1: 
✅ Good commit messages :+1: 
✅ Good descriptive PR :+1: 
✅ Well done :clap: you did all changes given by the previous reviewer :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**